application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -131,6 +151,86 @@ private RestAction<Message> sendExplanationMessage(GuildMessageChannel threadCha
         return action.setEmbeds(embeds);
     }
 
+    /**
+     * Determine between the title of the thread and the first message which to send to the AI. It
+     * uses a simple heuristic of length to determine if enough context exists in a question. If the
+     * title is used, it must also include a question mark since the title is often used more as an
+     * indicator of topic versus a question.
+     * 
+     * @param originalQuestion The first message of the thread which originates from the question
+     *        asker.
+     * @param threadChannel The thread in which the question was asked.
+     * @return An answer for the user from the AI service or a message indicating either an error or
+     *         why the message wasn't used.
+     */
+    RestAction<Message> constructChatGptAttempt(ThreadChannel threadChannel,
+            String originalQuestion) {
+
+        Optional<String> questionOptional = prepareChatGptQuestion(threadChannel, originalQuestion);
+        Optional<String> chatGPTAnswer;
+
+        String question = questionOptional.orElseThrow();
+        logger.debug("The final question sent to chatGPT: {}", question);
+        chatGPTAnswer = chatGptService.ask(question);
```

Comment:
What if the response is over the character limit? ChatGPT often gives really big responses, so I can't imagine this to not happen.

Reply:
Yup, that's already an issue with `/chatgpt` that is on master. It silently crashes if it's over the character limit.

And just cutting the answer to fit the limit is a bad solution, obviously. What's the point of getting 80% of the answer, if it's a code example or a step-by-step guide? Code won't work, guide is incomplete and useless.

So one solution is to prompt it to fit the answer within limits, and test it to make sure it works well. With a hard cutoff if a limit is reached.

Another would be to post lengthier answers as a file.

Reply:
Yeah, that's one of my proposed solutions.

We either tell it to fit the answer withing the limit (`limit - 500` for optimal results), with a fallback. Or we let it post the full answer as a file.

I like both of them. First one is short and neat, no files. My only worry is that the quality of the answers might lack. For example, can you fit a coherent guide how to install and setup something, with examples, within the limit.

Reply:
Tested few examples, it seems embed length is enough for most guides with examples.

My worry with embeds is that they are too narrow for code, it might ruin the formatting. Since code would be needed in almost every answer, it's something to consider.

And 2000 char limit or regular messages is likely not enough for bigger answers. Close, but not quite there.

Reply:
> Tested few examples, it seems embed length is enough for most guides with examples.
>My worry with embeds is that they are too narrow for code, it might ruin the formatting. Since code would be needed in almost every answer, it's something to consider.
>
>And 2000 char limit or regular messages is likely not enough for bigger answers. Close, but not quite there.

Thus, we should fit answers in a file?

Reply:
Try embeds first, for cosistency. Post a picture of a more complicated answer with code examples.

Files are the last restort, since they are kinda ugly and weird UI wise. You know, getting a file in response to a bot command.

Reply:
A file is fine for PC, you have good UX then, not as good as just a message but it's fine.

On phone, yeah uh, have fun opening notepad on your phone to read it.